react/week2/to_do_list/src/index.js

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const todos = [
+  {
+    id: 0,
+    description: "Get out of bed",
+    color: "brown",
+    deadline: "2020-06-08",
+    done: false,
+    Visibility: true
+  },
+  {
+    id: 1,
+    description: "Brush teeth",
+    color: "red",
+    deadline: "2020-06-08",
+    done: false,
+    Visibility: true
+  },
+  {
+    id: 2,
+    description: "Eat breakfast",
+    color: "orange",
+    deadline: "2020-06-08",
+    done: true,
+    Visibility: true
+  }
+];
+
+function getRandomColor() {
+  var letters = "0123456789ABCDEF";
+  var color = "#";
+  for (var i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+}
+
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { renderTrigger: true };
+  }
+
+  deleteTask(index) {
+    this.props.taskData[index].Visibility = false;
+    this.setState({ renderTrigger: !this.state.renderTrigger });
+  }
+
+  taggleDoneState(index) {
+    this.props.taskData[index].done = !this.props.taskData[index].done;
+    this.setState({ renderTrigger: !this.state.renderTrigger });
+  }
+
+  taskStyle(done) {
+    if (done) {
+      return { textDecoration: "line-through" };
+    }
+  }
+
+  addRandomTask = () => {
+    console.log(this.props.taskData);
+    this.props.taskData.push({
+      id: this.props.taskData.length,
+      description: "Random Text",
+      color: getRandomColor(),
+      deadline: "2020-12-29",
+      done: false,
+      Visibility: true
+    });
+    this.setState({ renderTrigger: !this.state.renderTrigger });
+  };
+
+  MakeMyList = () => {
```

Comment:
A component shouldn't be defined as a method within a component - it should simply be written as a new component. 

I would consider also consider renaming that component to `TodoList`. 

When naming components you should think in terms of what they are, not how it's implemented.